File: pbset2/readability/c/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");

    float letters = 0;
    float words = 1;
    float sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i ++)
    {
        char c = tolower(text[i]);
        if ('a' <= c && c <= 'z')
        {
            letters++;
        }
        if (c == ' ')
        {
            words++;
        }
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }
    float average_letters = (letters / words) * 100;
    float average_sentences = (sentences / words) * 100;
    int index = round(0.0588 * average_letters - 0.296 * average_sentences - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %i\n", index);
    return 0;
}

File: pbset2/readability/go/readability.go
package main

import (
	"fmt"
	"math"
	"unicode"
)

func main() {

}

func readability(text string) string {
	var words float64 = 1
	var letters, sentences float64
	for _, c := range text {
		c = unicode.ToLower(c)
		if unicode.IsLetter(c) {
			letters++
		}
		if unicode.IsSpace(c) {
			words++
		}
		if c == '.' || c == '!' || c == '?' {
			sentences++
		}
	}
	index := math.Round(0.0588*((letters/words)*100) - 0.296*((sentences/words)*100) - 15.8)

	switch {
	case index < 1:
		return "Before Grade 1"
	case index > 16:
		return "Grade 16+"
	default:
		return fmt.Sprintf("Grade %d", int(index))
	}
}
